File: 백준/Python/Silver/1002.터렛/터렛.py
import sys
input=sys.stdin.readline
n=int(input())
for _ in range(0,n):
    x1,y1,r1,x2,y2,r2=map(int,input().split())
    L2=(x1-x2)**2+(y1-y2)**2
    rR=(r1+r2)**2
    rr=(r1-r2)**2
    if L2==0 and r1==r2:
        print(-1)
    elif rR>L2>rr:
        print(2)
    elif rR==L2 or rr==L2:
        print(1)
    else:
        print(0)

File: 백준/C#/Silver/1002.터렛/터렛.cs
using System.Text;

namespace BOJ_C_sharp
{
    internal class Program_1002
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            StringBuilder sb = new StringBuilder();
            
            for (int i = 0; i < n; i++)
                {
                    string[] ss = Console.ReadLine().Split(' ', StringSplitOptions.RemoveEmptyEntries);
                    int x1 = int.Parse(ss[0]);
                    int y1 = int.Parse(ss[1]);
                    int r1 = int.Parse(ss[2]);
                    int x2 = int.Parse(ss[3]);
                    int y2 = int.Parse(ss[4]);
                    int r2 = int.Parse(ss[5]);

                    int dx = x1 - x2;
                    int dy = y1 - y2;
                    int L2 = dx * dx + dy * dy;
                    int rR = (r1 + r2) * (r1 + r2);
                    int rr = (r1 - r2) * (r1 - r2);

                    int result;
                    if (L2 == 0 && r1 == r2)
                        result = -1;
                    else if (rR > L2 && L2 > rr)
                        result = 2;
                    else if (rR == L2 || rr == L2)
                        result = 1;
                    else
                        result = 0;

                    sb.AppendLine(result.ToString());
                }

             Console.Write(sb.ToString());
        }
    }
}